src/panels/alarm-table/alarm_details.js

```diff
@@ -15,20 +26,175 @@ export class AlarmDetailsCtrl {
+          button = this.INCORRECT_FILLED;
+          break;
+        }
+      }
+    }
+    return button;
+  }
+
+  detailFeedbackOkayButton(reductionKey) {
+    let button = this.CORRECT_FILLED;
+    if (this.$scope.situationFeedback) {
+      for (let feedback of this.$scope.situationFeedback) {
+        if (feedback.alarmKey == reductionKey && feedback.feedbackType == "FALSE_POSITVE") {
+          button = this.CORRECT_OUTLINED;
+          break;
+        }
+      }
+    }
+    return button;
+  }
+
+  initalizeFeeback() {
+    let feedback = [];
+    for (let alarm of this.$scope.alarm.relatedAlarms) {
+      feedback.push(this.alarmFeedback(this.$scope.alarm.reductionKey, this.fingerPrint(this.$scope.alarm), alarm, "CORRECT", "ALL_CORRECT", this.contextSrv.user.login));
+    }
+    return feedback;
+  }
+
+  markIncorrect(reductionKey) {
+    for (let feedback of this.$scope.situationFeedback) {
+      if (feedback.alarmKey == reductionKey) {
+        feedback.feedbackType = "FALSE_POSITVE";
+        break;
+      }
+    }
+  }
+
+  markCorrect(reductionKey) {
+    for (let feedback of this.$scope.situationFeedback) {
+      if (feedback.alarmKey == reductionKey) {
+        feedback.feedbackType = "CORRECT";
+        break;
+      }
+    }
+  }
+
+  submitEditedFeedback(form) {
+    for (let feedback of this.$scope.situationFeedback) {
+      feedback.reason = form.reason;
+    }
+    this.submitFeedback(this.$scope.situationFeedback);
+  }
+
+  submitFeedback(feedback) {
+    let self = this;
+    this.getDatasource().then(ds => { return ds.submitSituationFeedback(self.$scope.alarm.id, feedback) })
+      .then(
+        function (response) {
+          self.$scope.editFeedback = false;
+          self.$scope.submittedFeedback = true;
+          self.$scope.hasSituationFeedback = true;
+        })
+      .catch(
+        function (reason) {
+          console.log("Got POST error: ", reason);
+          self.$scope.editFeedback = false;
+        });
+  }
+
+  updateFeedback(feedback) {
+    for (let fb of feedback) {
+      for (let ifb of this.$scope.situationFeedback) {
+        if (fb.alarmKey === ifb.alarmKey)
+          ifb = fb;
+          console.log("updated", ifb);
+      }
+    }
+  }
+
+  editSituationFeedback() {
+    this.$scope.editFeedback = true;
+    this.$scope.submittedFeedback = false;
+  }
+
+  situationFeedbackButton() {
+    let button = this.CORRECT_OUTLINED;
+    let fingerprint = this.fingerPrint(this.$scope.alarm);
+    if (this.$scope.situationFeedback) {
+      for (let feedback of this.$scope.situationFeedback) {
+        if (feedback.situationFingerprint == fingerprint && this.$scope.hasSituationFeedback) {
+          button = this.CORRECT_FILLED;
+          break;
+        }
+      }
+    }
+    return button;
+  }
+
+  alarmFeedback(situationKey, fingerprint, alarm, feedbackType, reason, user) {
+    let feedback = {};
```

Comment:
Can we use the `OnmsSituationFeedback` type here instead of redefining it as a map?

Reply:
done